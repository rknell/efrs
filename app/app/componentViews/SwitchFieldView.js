import React, { Component } from 'react'
import { Switch } from 'react-native'
import { observer } from 'mobx-react/native'
import FieldBase from './FieldBase'

@observer
class SwitchFieldView extends Component {
  render () {
    const settings = this.props.settings || { Label: this.props.label }
    const style = this.props.globalStyle || {}
    return (
      <FieldBase globalStyle={style}
        translate={this.props.translate}
        hidden={this.props.editMode ? false : settings['Hidden']}
        label={settings.Label}>
        <Switch trackColor={style.primaryColour} value={this.props.value} onValueChange={this.props.onChange} />
      </FieldBase>
    )
  }
}

export default SwitchFieldView
