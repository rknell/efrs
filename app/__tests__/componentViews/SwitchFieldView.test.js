import React from 'react'
import SwitchFieldView from '../../app/componentViews/SwitchFieldView'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

it('should render with no props', () => {
  const wrapper = shallow(<SwitchFieldView />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should render with settings', () => {
  const wrapper = shallow(<SwitchFieldView settings={{ Label: 'TEST' }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should be able to hide', () => {
  const wrapper = shallow(<SwitchFieldView settings={{ Hidden: true }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should render correctly in edit mode', () => {
  const wrapper = shallow(<SwitchFieldView editMode />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should take a non settings label', () => {
  const wrapper = shallow(<SwitchFieldView label='TEST LABEL' />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should take a globalstyle object and apply it', () => {
  const wrapper = shallow(<SwitchFieldView globalStyle={{ primaryColour: 'black' }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
