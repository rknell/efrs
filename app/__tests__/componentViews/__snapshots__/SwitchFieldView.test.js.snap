// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to hide 1`] = `
<FieldBase
  globalStyle={Object {}}
  hidden={true}
>
  <Switch />
</FieldBase>
`;

exports[`should render correctly in edit mode 1`] = `
<FieldBase
  globalStyle={Object {}}
  hidden={false}
>
  <Switch />
</FieldBase>
`;

exports[`should render with no props 1`] = `
<FieldBase
  globalStyle={Object {}}
>
  <Switch />
</FieldBase>
`;

exports[`should render with settings 1`] = `
<FieldBase
  globalStyle={Object {}}
  label="TEST"
>
  <Switch />
</FieldBase>
`;

exports[`should take a globalstyle object and apply it 1`] = `
<FieldBase
  globalStyle={
    Object {
      "primaryColour": "black",
    }
  }
>
  <Switch
    trackColor="black"
  />
</FieldBase>
`;

exports[`should take a non settings label 1`] = `
<FieldBase
  globalStyle={Object {}}
  label="TEST LABEL"
>
  <Switch />
</FieldBase>
`;
